docs/hugo/content/design/ADR-2023-02-Adoption-Policy.md

```diff
@@ -0,0 +1,258 @@
+| detach-on-delete   | detach-on-delete-if-exists | No                                | Manages the resource, detaches on delete |
+| skip               | detach-on-delete-if-exists | Yes                               | Skips managing the resource              |
+| skip               | detach-on-delete-if-exists | No                                | Skips managing the resource              |
+
+¹: Actual behavior depends on earlier sections of this design
+
+**Advantages:**
+- ???
+
+**Disadvantages:**
+- Users need to understand this complicated table to figure out what is actually happening to their resource, it's not
+  immediately obvious based on the annotations.
+- There's no way to tell which option was actually chosen for users.
+- We need to somehow remember which option we chose so we don't think that a resource we created needs to be adopted after
+  a crash or pod restart. That further increases the number of annotations used by this feature...
+
+##### Option 4.2: reconcile-policy-if-exists annotation
+
+A new property `RECONCILE_POLICY_IF_EXISTS` in the operator global secret, and an annotation on each
+resource `serviceoperator.azure.com/reconcile-policy-if-exists` would allow users to control this behavior.
+
+Possible `reconcile-policy-if-exists`'s are the same as the existing `reconcile-policy`
+- `manage`: Corresponding to scenario 1 above.
+- `detach-on-delete`: Corresponding to scenario 2 above.
+- `skip`: Corresponding to scenario 3 above.
+
+The implementation of `reconcile-policy-if-exists` would basically be a delayed defaulting of `reconcile-policy` applied
+when reconciliation ran for the first time. The operator would check if the resource already exists and write the corresponding 
+`reconcile-policy` annotation to the resource if the resource existed.
+
+**Advantages:**
+- Uses a concept we already have.
+
+**Disadvantages:**
+- Possibility of collision between user specified `reconcile-policy` and the `reconcile-policy` written by the operator because
+  of `reconcile-policy-if-exists`. 
+  - We can mitigate this issue by rejecting some configurations with a webhook, although we have to be careful how we craft
+    that webhook because the operator itself needs to be able to set both. We can either just block both on create but allow
+    both on update, or we can block both on create/update but only as long as the finalizer hasn't been set. Once the finalizer 
+    has been set, changing `reconcile-policy-if-exists` has no effect (we've already applied whatever we needed to) and all changes
+    to the policy should flow via the `reconcile-policy` annotation.
+- Excess events generated by us setting `reconcile-policy`, which is an annotation we watch for updates. We can deal with this: 
+  - If the `reconcile-policy-if-exists` is `skip` this is safe as we never re-reconcile when `reconcile-policy` is being set to skip.
```

Comment:
> we never re-reconcile when `reconcile-policy` is being set to skip.

Clarified by @matthchr on Slack, but this wasn't exactly clear to me that this referred to only the reconciliation where `skip` is set. Future periodic reconciliations still happen for `skip`-ped resources.

Reply:
Have clarified this in the design by rephrasing to:

> If the `reconcile-policy-if-exists` is `skip` this is safe as we don't have to trigger reconciliation if the only modification to the resource is that the `reconcile-policy` is being set to `skip`. This helps avoid a situation where we reconcile once, issue a GET to Azure, see that the resource exists and update the `reconcile-policy` annotation to be `skip`, which then triggers another reconciliation and another GET to Azure.